.github/workflows/e2e-nightly.yaml

```diff
@@ -0,0 +1,159 @@
+---
+name: E2E Nightly
+
+on:
+  workflow_dispatch:
+  schedule:
+    - cron: '30 5 * * *'
+  push: # TODO: remove me!
+    branches:
+      - nightly
+
+jobs:
+  deploy:
+    name: Deploy the operator
+    runs-on: 'ubuntu-latest'
+    env:
+      TAG: 3.5
+      CATALOG_PATH: ./bundle/quay-operator.catalogsource.yaml
+      OG_PATH: ./bundle/quay-operator.operatorgroup.yaml
+      SUBSCRIPTION_PATH: ./bundle/quay-operator.subscription.yaml
+      QUAY_SAMPLE_PATH: ./config/samples/managed.quayregistry.yaml
+      OPERATOR_PKG_NAME: quay-operator-test
+      NAMESPACE: flv-e2e
+    steps:
+      - name: Check out the repo
+        uses: actions/checkout@v2
+
+      - name: Setup catalog source yaml
+        uses: mikefarah/yq@master
+        env:
+          CATALOG_IMAGE: quay.io/fmissi/quay-operator-index:3.5
+        with:
+          cmd: |
+            yq e -i '
+              .spec.image = env(CATALOG_IMAGE) |
+              .metadata.name = env(OPERATOR_PKG_NAME)
+              ' ${CATALOG_PATH}
+
+      - name: Create catalog source
+        uses: actions-hub/kubectl@master
+        env:
+          KUBE_CONFIG: ${{ secrets.KUBE_CONFIG }}
+        with:
+          args: create -n openshift-marketplace -f ${{ env.CATALOG_PATH }}
+
+      - name: Setup operator group yaml
+        uses: mikefarah/yq@master
+        with:
+          cmd: yq e -i '.spec.targetNamespaces[0] = env(NAMESPACE)' ${OG_PATH}
+
+      - name: Create operator group
+        uses: actions-hub/kubectl@master
+        env:
+          KUBE_CONFIG: ${{ secrets.KUBE_CONFIG }}
```

Comment:
Fyi: Alternative to using OCP cluster, there is [kind](https://kind.sigs.k8s.io/) tool (came across this during olm time ([example](https://github.com/operator-framework/operator-lifecycle-manager/blob/2fd014019ca8cc86ae859da797e814b72c299257/.github/workflows/e2e-kind.yml#L16)) which lets you spin up cluster in <2min. We probably need to:
- install olm using the `operator-sdk` cli ([doc](https://sdk.operatorframework.io/docs/cli/operator-sdk_olm_install/))
- create `crds` on the cluster before hand
I see gatekeeper operator also does something similar [here](https://github.com/gatekeeper/gatekeeper-operator/blob/master/.github/workflows/olm_tests.yaml#L47-L65)
This reduces overhead on the OCP cluster and could get away with not having to cleanup resources

Also, noob me: Does quay even work on a non-openshift cluster?

Reply:
Yeah I heard of kind! This being an end-to-end test I think there is a point in running it against a proper openshift cluster. If/when we end up in openshift CI, then we will have the CI itself bootstrapping clusters for us. In the meanwhile we can use [kuttl](https://kuttl.dev) (credit to Jonathan for presenting it to the team) which I think could help us getting stuff setup/cleanup more easily.

And I'm a noob here too, but can't see why both quay and the operator wouldn't work in any k8s cluster as long as olm is installed(:

Reply:
Quay itself works fine on k8s, but with the Operator we need OpenShift as we rely on Route objects (which are OCP specific) for ingress.

Reply:
Oh! Does it work on [OKD](https://www.okd.io/) then?

Reply:
It should, OKD is a community supported version of OpenShift.

Reply:
I think you'll find it safer to stick with OCP itself, not sure how up to date OKD stays with OCP nightlies...

Reply:
Definitely @BillDett , I'm asking mostly out of curiosity ([and because our docs seem to be misleading about this](https://github.com/quay/quay-operator#getting-started), perhaps it has worked previously 🤔